kong/tracing/propagation.lua

```diff
@@ -397,6 +411,63 @@ local function parse_aws_headers(aws_header)
   return trace_id, span_id, should_sample
 end
 
+local function parse_gcp_headers(gcp_header)
+  local warn = kong.log.warn
+
+  if type(gcp_header) ~= "string" then
+    return nil, nil, nil
+  end
+
+  local trace_id, span_id, should_sample = match(gcp_header, GCP_TRACECONTEXT_PATTERN)
+
+  -- GCP header pattern is lenient on field existence and length
+  -- coalesce empty values to nil for easier validation
+  if trace_id == "" then
+    trace_id = nil
+  end
```

Comment:
Nitpick: you **could** do
`trace_id = trace_id ~= "" and trace_id`

and later check for `falsy` instead of `nil`:
`if not trace_id or ...`

same for the other fields.
But it's really just personal preference, feel free to disregard this one.